src/features/eighteenBirthday/pages/components/EighteenBirthdayCard.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { act, fireEvent, render } from '@testing-library/react-native'
+import { rest } from 'msw'
+import React, { RefObject } from 'react'
+import Swiper from 'react-native-web-swiper'
+
+import { navigate } from '__mocks__/@react-navigation/native'
+import { GetIdCheckTokenResponse, UserProfileResponse } from 'api/gen'
+import { AuthContext } from 'features/auth/AuthContext'
+import { env } from 'libs/environment'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
+import { server } from 'tests/server'
+import { flushAllPromisesTimes } from 'tests/utils'
+import { SnackBarHelperSettings } from 'ui/components/snackBar/types'
+
+import { EighteenBirthdayCard } from './EighteenBirthdayCard'
+
+const email = 'email@domain.ext'
+const firstName = 'Jean'
+const token = 'XYZT'
+const mockDisplayInfosSnackBar = jest.fn()
+
+jest.mock('ui/components/snackBar/SnackBarContext', () => ({
+  useSnackBarContext: () => ({
+    displayInfosSnackBar: jest.fn((props: SnackBarHelperSettings) =>
+      mockDisplayInfosSnackBar(props)
+    ),
+  }),
+}))
+
+describe('EighteenBirthdayCard', () => {
+  it('should render eighteen birthday card', async () => {
+    jest.useFakeTimers()
+    const firstTutorial = await renderEighteenBirthdayCard()
+    jest.advanceTimersByTime(2000)
+
+    expect(firstTutorial).toMatchSnapshot()
+    jest.useRealTimers()
+  })
+
+  it('should go to id check when user is authed', async () => {
+    simulateAuthedUser()
+    simulateIdCheckToken()
+    const { getByText } = await renderEighteenBirthdayCard()
+
+    fireEvent.press(getByText('Verifier mon identité'))
+    expect(navigate).toBeCalledWith('IdCheck', { email, licenceToken: token })
+  })
+
+  it('should go to login check when user is not authed', async () => {
+    const { getByText } = await renderEighteenBirthdayCard({
+      isLoggedIn: false,
+    })
+
+    fireEvent.press(getByText('Verifier mon identité'))
+    expect(navigate).toBeCalledWith('Login')
+    expect(mockDisplayInfosSnackBar).toBeCalledWith({
+      message: `Tu n'es pas connecté !`,
+    })
+  })
+})
+
+async function renderEighteenBirthdayCard({ isLoggedIn } = { isLoggedIn: true }) {
+  const ref = { current: { goToNext: jest.fn() } }
+  const renderAPI = render(
+    reactQueryProviderHOC(
+      <AuthContext.Provider value={{ isLoggedIn, setIsLoggedIn: jest.fn() }}>
+        <EighteenBirthdayCard
+          activeIndex={0}
+          index={0}
+          lastIndex={0}
+          swiperRef={(ref as unknown) as RefObject<Swiper>}
+        />
+      </AuthContext.Provider>
+    )
+  )
+  await act(() => flushAllPromisesTimes(10))
```

Comment:
Tu n'as pas de warning sans cela ?
```suggestion
  await act(async () => await flushAllPromisesTimes(10))
```

Reply:
OK